### PR TITLE
Add tests for join and aggregation related to NaN handling

### PR DESCRIPTION
**Pull request by bikramSingh91**
Summary:
Hash join and hash aggregation employ data structures like
VectorHasher, RowContainer and ContainerRowSerde apart from the usual
BaseVector to implement a custom hash table and store intermediate
state. All these have been updated in #9963 to handle NaN values
where all binary representations of the same are considered equal
and hash to the same value. Therefore, this change only adds unit
tests to verify the expected NaN behavior.

Differential Revision: D58365384

### CODE REVIEW

File: velox/exec/tests/HashJoinTest.cpp [hunk 1]
```diff
@@ -7797,4 +7797,38 @@ DEBUG_ONLY_TEST_F(HashJoinTest, spillCheckOnLeftSemiFilterWithDynamicFilters) {
...
 }
+
+TEST_F(HashJoinTest, nanKeys) {
+  // Verify the NaN values with different binary representations are considered
+  // equal.
+  static const double kNan = std::numeric_limits<double>::quiet_NaN();
+  static const double kSNaN = std::numeric_limits<double>::signaling_NaN();
+  auto probeInput = makeRowVector(
+      {makeFlatVector<double>({kNan, kNan}), makeFlatVector<int64_t>({1, 2})});
+  auto buildInput = makeRowVector({makeFlatVector<double>({kNan, 1})});
```
Comment (kagamiori) [inline velox/exec/tests/HashJoinTest.cpp:7808]: I think you meant to use kSNaN in probe or build input?

File: velox/exec/tests/HashJoinTest.cpp [hunk 2]
```diff
@@ -7797,4 +7797,38 @@ DEBUG_ONLY_TEST_F(HashJoinTest, spillCheckOnLeftSemiFilterWithDynamicFilters) {
...
+                  .hashJoin(
+                      {"t0"},
+                      {"u0"},
+                      PlanBuilder(planNodeIdGenerator)
+                          .values({buildInput})
+                          .project({"c0 AS u0"})
+                          .planNode(),
+                      "",
+                      {"t0", "u0", "t1"},
+                      core::JoinType::kLeft)
```
Comment (kagamiori) [inline velox/exec/tests/HashJoinTest.cpp:7823]: Do we need to test other join types? Do they all use the same comparison code path?

Comment (bikramSingh91) [reply velox/exec/tests/HashJoinTest.cpp:7823]: They use the same data structures as mentioned in the summary.

**Review by kagamiori — APPROVED** [review]
LGTM. Thanks for adding the test coverage!

**Pull request closed without merging**